Enhanced the syntax support (WIP)

zealot128 opened the pull request:
Recently, I've switched to treesitter with Neovim, but found out, that Pug support is not included, yet. Then I found for repo which is a good start! 

I added the following syntax that i use regularly:

- inner pipe content: 

```
div
  | bar
```

- inline js:

```
div(:foo='inlineJsHere()' v-if='AndHere()')
  | inline js: {{ 1 + 1 === 2 }}
```

- wrapped tags:

```
div.bar: div.foo: div.blub Finally content here
```

zealot128 committed:
Fix: upgrade tree-sitter generate output

zealot128 committed:
Feat: Enhance parser rules

- Multi line pipe blocks support
- wrapped tags: div: div Foo
- inline JS (non Multiline)
- Added vue-attributes
- refac rules

zealot128 committed:
CI: Add Github Action, Readme, missing source files

bindings/*, parser.c must be checked in, too, otherwise GH Actions cannot build

zealot128 commented:
Added:

- Github action for test
- much more syntax, see corpus

zealot128 committed:
Feat: Doctype + Multiline content with dot

RianFuro commented:
Hey, thanks for sharing!

I'll gladly take your contribution, however I'd like to note that I'm not actively maintaining the code; I really just made this as a one-off to play around with neovims treesitter integration.

One thing that damped my motivation was how the vue-parser at the time handled the html-markups; I ended up using the html parser for vue files and used injections instead. May I ask how you went about that? Or does the vue parser now better accomodate different languages?

Also, I see you've replaced the `package-lock` with a yarn-lockfile - any reason for that besides preference?

zealot128 commented:
@RianFuro 

I also forked the Vue-treesitter and submitted a PR which changes the content of the template block to "raw text" if there is any kind of attribute. Unfortunately, no response from the maintainer yet.
https://github.com/ikatyang/tree-sitter-vue/pull/13

I am currently running that fork of vue-treesitter + my fork/pr + [a fork from my of nvim with the pug config](https://github.com/nvim-treesitter/nvim-treesitter/pull/1549) without any problems and happily coding vue + pug files :)

If the vue-treesitter would be merged, that would open up the merge for the nvim-treesitter config and it could finally land in mainline nvim-treesitter...

Regarding package-lock, no specific reasoning, I usually use yarn(1) all the time. I could change it, but if you don't intend to maintain the package anyhow, I could also keep my fork.

theHamsta committed:
fix: try to avoid segfaults in scanner.cc

I think the main change is checking for lexer->lookahead before doing
lexer->get_column(lexer) since it was segfaulting in exactly that
function.

zealot128 committed:
Merge pull request #1 from theHamsta/avoid-segfault

fix: try to avoid segfaults in scanner.cc

zealot128 committed:
Fix: Fix most of tests by setting associativity

mclaughlinconnor committed:
Feat: angular square bracket attributes

mclaughlinconnor committed:
Feat: angular outputs with javascript

mclaughlinconnor committed:
Feat: angular directives

mclaughlinconnor committed:
Feat: unbuffered javascript code support

mclaughlinconnor committed:
Fix: injections not working with new javascript

mclaughlinconnor committed:
Fix: remove unused $ variables

mclaughlinconnor committed:
Feat: add conditionals

mclaughlinconnor committed:
Fix: actually include injections

mclaughlinconnor committed:
Fix: make all javascript nodes look like $.javascript

mclaughlinconnor committed:
Feat: add non-angular javascript attributes for non-functions only

mclaughlinconnor committed:
Fix: fix not handling tags inside of unbuffered code properly

mclaughlinconnor committed:
Feat: add buffered code

mclaughlinconnor committed:
Feat: unescaped buffered code

mclaughlinconnor committed:
Feat: case statements

mclaughlinconnor committed:
Fix: fix pipes not being treated as root nodes, also gave better name

mclaughlinconnor committed:
Feat: add code attributes for non-Angular attributes

mclaughlinconnor committed:
Fix: add missing javascript alias for angular interpolations

mclaughlinconnor committed:
Fix: fix first character of attribute not being included in $.javascript

mclaughlinconnor committed:
Feat: allow one level of function calls in pug javascript attributes

mclaughlinconnor committed:
Feat: add pug javascript interpolation

mclaughlinconnor committed:
Fix: allow hash tags and brackets in tag content

mclaughlinconnor committed:
Fix: allow dot being on newline for content

mclaughlinconnor committed:
Feat: add javascript inside of script dot blocks

mclaughlinconnor committed:
Fix: fix dedent not being handled on end of unbuffered code breaking comments

mclaughlinconnor committed:
Fix: strip trailing whitespace from source

mclaughlinconnor committed:
Fix: fix conditionals in tags sometimes being booted to the root

mclaughlinconnor committed:
Fix: fix comments following unbuffered js breaking

mclaughlinconnor committed:
Fix: fix conditionals being merged together with errors

mclaughlinconnor committed:
Docs: add todos for angular templates and standalone class components

mclaughlinconnor committed:
Feat: support angular template attributes

mclaughlinconnor committed:
Fix: remove useless precedence on tag

mclaughlinconnor committed:
Fix: fix conditional statements being a bit enthusiastic about how much they capture

mclaughlinconnor committed:
Fix: fix case statements not finishing their captures in the right place

mclaughlinconnor committed:
Fix: fix comments starting after the first line being limited to one line

mclaughlinconnor committed:
Docs: add many more todos

mclaughlinconnor committed:
Fix: fix consecutive dedents being merged together

mclaughlinconnor committed:
Fix: remove commented code from scanner

mclaughlinconnor committed:
Lint: reformat $.children across multiple lines

mclaughlinconnor committed:
Docs: the $._dedent is absolutely needed here

mclaughlinconnor committed:
Feat: parse Angular attributes with dots in them properly

mclaughlinconnor committed:
Build: add log.html to gitignore

mclaughlinconnor committed:
Fix: fix broken pipe test with expected failure

mclaughlinconnor committed:
Fix: remove useless injection queries

mclaughlinconnor committed:
Feat: add self-closing tags

mclaughlinconnor committed:
Feat: add filters

mclaughlinconnor committed:
Docs: update readme with current state

mclaughlinconnor committed:
Docs: remove todos fixed by repairing dedents

mclaughlinconnor committed:
Feat: add template inheritance

mclaughlinconnor committed:
Lint: remove a few extra unused $'s

mclaughlinconnor committed:
Feat: add mixins

mclaughlinconnor committed:
Feat: add iterators

mclaughlinconnor committed:
Fix: remove colons from tag name supported characters

mclaughlinconnor committed:
Fix: allow slashes and other non-alphanumerics in extend filenames

mclaughlinconnor committed:
Feat: add includes

mclaughlinconnor committed:
Chore: add keyword alias to all keywords to prepare for highlighting

mclaughlinconnor committed:
Fix: fix comments not ending after dedents

mclaughlinconnor committed:
Fix: fix newlines inside dot continues causing the next dedent to be missed

mclaughlinconnor committed:
Chore: bump tree-sitter-cli to 0.20.7

mclaughlinconnor committed:
Feat: add syntax highlighting queries

mclaughlinconnor committed:
Fix: add angular two way binding support

mclaughlinconnor committed:
Docs: add todo for angular template directive bindings

mclaughlinconnor committed:
Docs: add some thoughts about nested function calls in pug js attributes

mclaughlinconnor committed:
Docs: add todo to rework regexes

mclaughlinconnor committed:
Docs: reword readme to be more inclusive of non-angular developers

mclaughlinconnor committed:
Merge pull request #6 from mclaughlinconnor/readme_reword

Docs: reword readme to be more inclusive of non-angular developers

zealot128 committed:
Merge pull request #4 from Pug-Angular-Editor-Tools/master

Further enhanced the syntax support

zealot128 committed:
Merge pull request #5 from Pug-Angular-Editor-Tools/bananabox

Fix Angular two way bindings
